cms/djangoapps/contentstore/tests/utils.py

```diff
@@ -299,8 +299,14 @@ def assertCoursesEqual(self, course1_id, course2_id):
                 store = self.store._get_modulestore_for_courselike(course2_id)  # pylint: disable=protected-access
                 new_name = 'course' if isinstance(store, SplitMongoModuleStore) else course2_item_loc.run
                 course2_item_loc = course2_item_loc.replace(name=new_name)
-            course2_item = self.store.get_item(course2_item_loc)
-
+                course2_item = self.store.get_item(course2_item_loc)
+                # The EnrollmentTrackuserPartition stores the course_id, which will vary split vs. old mongo.
+                # Also at least one test calls this comparison method with a purposefully different course.
+                # Clear stored course_id so that own_metadata call below does not fail.
+                course1_item.user_partitions[0].parameters['course_id'] = None
```

Comment:
The problem is that we need the course_id to return the groups from the user partition (and groups is a property). I don't see how we can pass the course_id unless I change groups completely in a method.

UserPartitions (after initial construction) are constructed via the from_json method. We don't have the course_id available then either.